client/state/comments/reducer.js

```diff
@@ -0,0 +1,176 @@
+/**
+ * External dependencies
+ */
+import Immutable from 'immutable';
+import { combineReducers } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import {
+	COMMENTS_RECEIVE,
+	COMMENTS_REMOVE,
+	COMMENTS_ERROR,
+	COMMENTS_COUNT_RECEIVE,
+	COMMENTS_LIKE,
+	COMMENTS_LIKE_UPDATE,
+	COMMENTS_UNLIKE,
+	COMMENTS_REQUEST,
+	COMMENTS_REQUEST_SUCCESS,
+	COMMENTS_REQUEST_FAILURE,
+	DESERIALIZE,
+	SERIALIZE,
+} from '../action-types';
+import {
+	createCommentTargetId,
+	listFindAndModify
+} from './utils';
+import {
+	PLACEHOLDER_STATE
+} from './constants';
+
+/***
+ * Creates a function that updates like count
+ * @param {Boolean} like true to like, false to unlike
+ * @returns {Function} function that updates like count for comment
+ */
+function getLikeSetter( { like = true, likeCount } ) {
+	return ( comment ) => comment.set( 'i_like', like )
+								.update( 'like_count', ( ct ) => {
+									if ( typeof likeCount !== 'undefined' ) {
+										return likeCount;
+									}
+
+									return like ? ct + 1 : ct - 1;
+								} );
+}
+
+/***
+ * Updates a specific state in the general state
+ * @param {Immutable.Map} state general state for all commentTargetId
+ * @param {Object} action action object
+ * @param {Number} action.siteId site identifier
+ * @param {Number} action.postId post identifier
+ * @param {Function|Object} updaterOrValue an updater function or a value
+ * @returns {Immutable.Map} new state
+ */
+function updateSpecificState( state, action, updaterOrValue ) {
```

Comment:
The reason I've done it like that is because I didn't want to call `createCommentTargetId( action.siteId, action.postId )` in every reducer and capture the logic of getting the specific state (and the generation of it's ID) in a single place, It is after all a helper function =)